docs/sources/flow/reference/components/loki.source.api.md

```diff
@@ -0,0 +1,82 @@
+---
+title: loki.source.api
+---
+
+# loki.source.api
+
+`loki.source.api` listens for HTTP requests containing Loki log entries and forwards them to other components capable
+of receiving log entries.
+
+The HTTP API exposed is compatible with [Loki push API][loki-push-api], using Loki's `logproto` format. This means that
+other [`loki.write`][loki.write] components can be used as a client and send requests to `loki.source.api`. This enables
+Grafana Agents to send log entries over the network to other Grafana Agents.
+
+[loki.write]: {{< relref "./loki.write.md" >}}
+[loki-push-api]: https://grafana.com/docs/loki/latest/api/#push-log-entries-to-loki
+
+## Usage
+
+```river
+loki.source.api LABEL {
+    http_address = "LISTEN_ADDRESS"
+    http_port = PORT
+    forward_to = RECEIVER_LIST
+}
+```
+
+## Arguments
+
+`loki.source.api` supports the following arguments:
+
+ Name                     | Type                 | Description                                                | Default | Required 
+--------------------------|----------------------|------------------------------------------------------------|---------|----------
+ `http_port`              | `int`                | The host port for the HTTP server to listen on.            | `0`     | yes      
+ `http_address`           | `string`             | The host address for the HTTP server to listen on.         | `""`    | yes      
+ `forward_to`             | `list(LogsReceiver)` | List of receivers to send log entries to.                  |         | yes      
+ `use_incoming_timestamp` | `bool`               | Whether or not to use the timestamp received from request. | `false` | no       
+ `labels`                 | `map(string)`        | The labels to associate with each received logs record.    | `{}`    | no       
+ `relabel_rules`          | `RelabelRules`       | Relabeling rules to apply on log entries.                  | `{}`    | no       
+
+The `relabel_rules` field can make use of the `rules` export value from a
+[`loki.relabel`][loki.relabel] component to apply one or more relabeling rules to log entries
+before they're forwarded to the list of receivers in `forward_to`.
+
+[loki.relabel]: {{< relref "./loki.relabel.md" >}}
+
+## Exported fields
+
+`loki.source.api` does not export any fields.
+
+## Component health
+
+`loki.source.api` is only reported as unhealthy if given an invalid configuration.
+
+## Metrics
```

Comment:
There are more metrics that come with the weaveworks' server. I didn't think they are important enough to be on this list. Some other components don't mention them at all.

The metrics I've skipped:
```
# HELP loki_source_api_inflight_requests Current number of inflight requests.
# HELP loki_source_api_tcp_connections_limit The max number of TCP connections that can be accepted (0 means no limit).
```

Happy to try unify these, but again, this could be something for the Server unification.